File: src/Models/Buildable/Installable/Installable.java
package Models.Buildable.Installable;

import Graphics.GraphicsList;
import Models.Buildable.Material.MaterialList;
import Models.Measurement;

public interface Installable {
    /**
     *
     * @return - A copy of the {@link Measurement} of the width of this Installable
     */
    Measurement totalWidth();

    /**
     *
     * @return - A copy of the {@link Measurement} of the height of the installable
     */
    Measurement totalHeight();

    /**
     *
     * @return - A {@link MaterialList} of {@link Models.Buildable.Material.Material} used to create this Installable
     */
    MaterialList materialList();

    /**
     *
     * @return - A {@link GraphicsList} of {@link Graphics.GraphicsInstructions} used to draw this Installable
     */
    GraphicsList graphicsList();
}

File: src/main/kotlin/Models/buildable/installable/Installable.kt
package models.buildable.installable

import graphics.GraphicsList
import models.buildable.material.MaterialList
import models.Measurement

interface Installable {
    /**
     *
     * @return - A copy of the [Measurement] of the width of this Installable
     */
    fun totalWidth(): Measurement

    /**
     *
     * @return - A copy of the [Measurement] of the height of the installable
     */
    fun totalHeight(): Measurement

    /**
     *
     * @return - A [MaterialList] of [models.buildable.material.Material] used to create this Installable
     */
    fun materialList(): MaterialList

    /**
     *
     * @return - A [GraphicsList] of [graphics.GraphicsInstructions] used to draw this Installable
     */
    fun graphicsList(): GraphicsList
}